include/zephyr/bluetooth/audio/audio.h

```diff
@@ -41,6 +41,18 @@ extern "C" {
 
 #define BT_AUDIO_BROADCAST_CODE_SIZE             16
 
+#ifndef POPCOUNT
+#define POPCOUNT(x) \
+({ \
+	uint8_t cnt = 0U; \
+	while (x != 0) { \
+		cnt += x & 1U; \
+		x >>= 1; \
+	} \
+	cnt; \
+})
+#endif
```

Comment:
We shouldn't define such a macro in `audio.h`.

If you want to support compilers that do not implement POPCOUNT, then we should probably add it somewhere in https://github.com/zephyrproject-rtos/zephyr/tree/main/include/zephyr/toolchain

Reply:
Is it OK to define this macro in 'common.h'?
https://github.com/zephyrproject-rtos/zephyr/tree/main/include/zephyr/toolchain/common.h

Reply:
i believe POPCOUNT should already be safe to use as-is for Zephyr's supported compilers? It's used in a few places in-tree (including in Bluetooth Mesh), and e.g `toolchain/gcc.h` already provides a mapping to `__builtin_popcount(x)`.